core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala

```diff
@@ -0,0 +1,216 @@
+      val (out4, err4) = executeCommand(incre_backupcommand)
+
+      if (!out4.contains("successful")) {
+        throw new Exception(s"Could not take successful backup")
+      }
+      stopCluster()
+      val (out5, err5) = executeCommand(s"rm -rf $snappyHome/work")
+      if (err5 != null && err5.length > 0) {
+        throw new Exception(s"Failed to remove work dir")
+      }
+
+      debugWriter.println(s"backup dir  = $backupDir")
+      // Find the latest two restore scripts
+      // val backupDirFile = new File(backupDir)
+      val backupDirs = backupDir.listFiles()
+      assert(backupDirs.length == 2)
+
+      val dir1 = new File(backupDir.getAbsolutePath, backupDirs(0).getName)
+      val dir2 = new File(backupDir.getAbsolutePath, backupDirs(1).getName)
+
+      var lastbackDir: File = null
+      if (dir2.lastModified() > dir1.lastModified()) {
+        lastbackDir = dir2
+      }
+      else {
+        lastbackDir = dir1
+      }
+
+      debugWriter.println(s"lastBackDir abs path = ${lastbackDir.getAbsolutePath}")
+
+      val (out6, err6) = executeCommand(s"find ${lastbackDir.getAbsolutePath} -iname restore.sh")
+
+      val restoreCmnds2 = out6.split("\n")
+      assert(restoreCmnds2.length == 2, "expected 2 restore commands")
+
+      debugWriter.println(s"after incre restore1  = ${restoreCmnds2(0)}")
+      debugWriter.println(s"after incre restore2  = ${restoreCmnds2(1)}")
+
+      assert(restoreCmnds2(0).contains("restore.sh") && restoreCmnds(1).contains("restore.sh"))
+
+      executeCommand(restoreCmnds2(0))
+      executeCommand(restoreCmnds2(1))
+
+      startupCluster()
+
+      val conn2 = getJdbcConnection(1527)
+      val stmnt2 = conn2.createStatement()
+      assert(stmnt2.execute("select * from test_app"))
+      val rs11 = stmnt2.getResultSet
+      cnt = 0
+      while (rs11.next()) {
+        cnt = cnt + 1
+        val v1 = rs11.getInt(1)
+        val v2 = rs11.getInt(2)
+        assert(v1 === 1 || v1 === 2 || v1 === 5 || v1 === 6 || v1 === 100 || v1 === 200)
+      }
+      assert(cnt === 6)
+
+      assert(stmnt2.execute("select * from testDD"))
+      val rs22 = stmnt2.getResultSet
+      cnt = 0
+      while (rs22.next()) {
+        cnt = cnt + 1
+        val v1 = rs22.getInt(1)
+        val v2 = rs22.getInt(2)
+        assert(v1 === 1 || v1 === 2 || v1 === 5 || v1 === 6 || v1 === 100 || v1 === 200)
+      }
+      assert(cnt === 6)
+    } finally {
+      debugWriter.close()
+      executeCommand(s"rm -rf $snappyHome/backup*")
+    }
+  }
+
+  def getJdbcConnection(netPort: Int): Connection = {
+    val driver = "io.snappydata.jdbc.ClientDriver"
+    Utils.classForName(driver).newInstance
+    var url: String = "jdbc:snappydata://localhost:" + netPort + "/"
+    DriverManager.getConnection(url)
+  }
+
+  // CWD will be assumed the same for all command which is $snappyHome
+  def executeCommand(command: String): (String, String) = {
```

Comment:
executeCommand() method can be moved to SnappyTestRunner for code reuse. Same is the case with getJdbcConnection()